slu/tests/test_contextual_slu/test.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+import os
+from tqdm import tqdm
+import yaml
+from slu.utils.config import YAMLPromptConfig
+from slu.dev.prompt_setup import setup_prompts, fill_nls_col
+from slu import constants as const
+import argparse
+
+
+def load_tests(path)-> dict:
+    """
+    Custom function to load test cases (.yaml file)
+    """
+    if not os.path.exists(path):
+        raise Exception(
+            f"Invalid path: {path}"
+        )
+    with open(path, 'r') as fp:
+        test_cases = yaml.load(fp, Loader=yaml.FullLoader)
+    
+    return test_cases
+
+def reset_output(dest)-> None:
+    """
+    Custom function to delete output files. 
+    """
+    if os.path.exists(os.path.join(dest,'prompts.yaml')):
+        os.remove(os.path.join(dest,'prompts.yaml'))
+        
+    if os.path.exists(os.path.join(dest,'missing_prompts.yaml')):
+        os.remove(os.path.join(dest,'missing_prompts.yaml'))
+        
+
+@pytest.mark.parametrize("test_case", load_tests(path="tests/test_contextual_slu/test_cases.yaml"))
+def test_prompt_config(test_case)-> None:
+    if test_case['type'] == 'prompt_config':
```

Comment:
Ideally each `if` should be a separate test. please take care of this from next time.